Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -7247,6 +7274,1965 @@ def panorama_install_file_content_update_command(args: dict):
+            summary_data.append(dataclass_from_element(device, ShowSystemInfoSummaryData,
+                                                       response.find("./result/system")))
+
+        return ShowSystemInfoCommandResult(
+            result_data=result_data,
+            summary_data=summary_data
+        )
+
+    @staticmethod
+    def get_available_software(topology: Topology,
+                               device_filter_str: Optional[str] = None) -> SoftwareVersionCommandResult:
+        """Get all available software updates"""
+        summary_data = []
+        for device in topology.all(filter_string=device_filter_str):
+            device.software.check()
+            for version_dict in device.software.versions.values():
+                summary_data.append(dataclass_from_dict(device, version_dict, SoftwareVersion))
+
+        return SoftwareVersionCommandResult(
+            summary_data=summary_data
+        )
+
+    @staticmethod
+    def download_software(topology: Topology, version: str,
+                          sync: bool = False, device_filter_str: Optional[str] = None) -> DownloadSoftwareCommandResult:
+        """Download the given software version to the device. This is an async command, and returns
+        immediately."""
+        result = []
+        for device in topology.all(filter_string=device_filter_str):
+            device.software.download(version, sync=sync)
+            result.append(GenericSoftwareStatus(
+                hostid=resolve_host_id(device),
+                started=True
+            ))
+
+        return DownloadSoftwareCommandResult(
+            summary_data=result
+        )
+
+    @staticmethod
+    def install_software(topology: Topology, version: str,
+                         sync: bool = False, device_filter_str: Optional[str] =None) -> InstallSoftwareCommandResult:
+
+        result = []
+        for device in topology.all(filter_string=device_filter_str):
+            device.software.install(version, sync=sync)
+            result.append(GenericSoftwareStatus(
+                hostid=resolve_host_id(device),
+                started=True
+            ))
+
+        return InstallSoftwareCommandResult(
+            summary_data=result
+        )
+
+    @staticmethod
+    def reboot(topology: Topology, hostid: str) -> RestartSystemCommandResult:
+
+        result = []
+        for device in topology.all(filter_string=hostid):
+            device.restart()
+            result.append(GenericSoftwareStatus(
+                hostid=resolve_host_id(device),
+                started=True
+            ))
+
+        return RestartSystemCommandResult(
+            summary_data=result
+        )
+
+    @staticmethod
+    def commit(topology: Topology, device_filter_string: Optional[str] = None) -> List[CommitStatus]:
+
+        result = []
+        for device in topology.active_devices(device_filter_string):
+            job_type = "Commit"
```

Comment:
what is it necessary to define that variable inside the for-loop? can't it be outside the for-loop? if its just used once in the function, then there isn't a reason to keep that variable, you can just remove it and do ```commit_type='Commit'```

Reply:
Agreed, fixed